pimd/pim_oil.c

```diff
@@ -146,6 +146,43 @@ struct channel_oil *pim_find_channel_oil(struct pim_instance *pim,
 	return c_oil;
 }
 
+void pim_channel_oil_change_iif(struct pim_instance *pim,
+				struct channel_oil *c_oil,
+				int input_vif_index,
+				const char *name)
+{
+	int old_vif_index = c_oil->oil.mfcc_parent;
+	struct prefix_sg sg = {.src = c_oil->oil.mfcc_mcastgrp,
+			       .grp = c_oil->oil.mfcc_origin};
+
+	if (c_oil->oil.mfcc_parent == input_vif_index) {
+		if (PIM_DEBUG_MROUTE_DETAIL)
+			zlog_debug("%s(%s): Existing channel oil %pSG4 already using %d as IIF",
+				   __PRETTY_FUNCTION__, name, &sg,
+				   input_vif_index);
+
+		return;
+	}
+
+	if (PIM_DEBUG_MROUTE_DETAIL)
+		zlog_debug("%s(%s): Changing channel oil %pSG4 IIF from %d to %d installed: %d",
+			   __PRETTY_FUNCTION__, name, &sg,
+			   c_oil->oil.mfcc_parent, input_vif_index,
+			   c_oil->installed);
+
+	c_oil->oil.mfcc_parent = input_vif_index;
+	if (c_oil->installed) {
+		if (input_vif_index == MAXVIFS)
+			pim_mroute_del(c_oil, name);
+		else
+			pim_mroute_add(c_oil, name);
+	} else
+		if (old_vif_index == MAXVIFS)
```

Comment:
why is this check needed?

Reply:
was trying to keep the code honest in that if we are not installed( that means old_vif_index was MAXVIFS.  So probably not needed 100%

Reply:
that's fair. i don't have a strong opinion one way or the other so maybe leave it as is.